### PR TITLE
[pointer_interceptor] Migrate to null safety

### PR DESCRIPTION
**Pull request by mzdm**
Related tracking issue https://github.com/flutter/flutter/issues/75567

also:
- Added package links to package list to main README.md
- Added `--web-renderer=html` to integration test command 
(seems to be failing without that)
- Not sure what to return in case of Shim methods 
(methods from engine are non-nullable so I replaced `null` with `an empty string`)

### CODE REVIEW

File: packages/pointer_interceptor/pubspec.yaml
Comment (stuartmorgan) [inline packages/pointer_interceptor/pubspec.yaml:3]: Once https://github.com/flutter/packages/pull/289 lands this can just be `0.9.0`, so we don't have to do a second pass over this package later.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
@ditman should do the full review here.

**Comment by ditman** [comment]
Thanks for the ping! This is my next task, I'll take a look within ~24h :)

File: packages/pointer_interceptor/CHANGELOG.md
Comment (ditman) [inline packages/pointer_interceptor/CHANGELOG.md:1]:
```suggestion
## 0.9.0
```

File: packages/pointer_interceptor/example/README.md
Comment (ditman) [inline packages/pointer_interceptor/example/README.md:11]: I checked the tests again, and this is an OK change.

The test is written to inspect the HTML code of the page, and the way views are rendered is slightly different between the html and canvaskit renderers.

In this case, `html` is the way to go. I'll try to figure out if there's a way to get canvaskit coverage as well, so this can be tested with both renderers!

File: packages/pointer_interceptor/example/lib/src/shim/dart_ui_fake.dart
Comment (ditman) [inline packages/pointer_interceptor/example/lib/src/shim/dart_ui_fake.dart:25]: This is fine. The Shim code should actually never be called, this exists here to appease the analyzer about web-only methods in the dart:ui library.

File: packages/pointer_interceptor/example/pubspec.yaml
Comment (ditman) [inline packages/pointer_interceptor/example/pubspec.yaml:7]:
```suggestion
  sdk: ">=2.12.0-0 <3.0.0"
  flutter: ">=1.26.0-0" # For integration_test from sdk
```

File: packages/pointer_interceptor/pubspec.yaml
Comment (ditman) [inline packages/pointer_interceptor/pubspec.yaml:3]:
```suggestion
version: 0.9.0
```

**Review by ditman — APPROVED** [review]
This looks great to me. I left some minor comments on the PR, and reversioned to 0.9.0. I've also migrated the integration_test code to null-safety!

Let's wait until #289 lands to merge this!

**Comment by ditman** [comment]
@googlebot I consent.

**Comment by ditman** [comment]
(The `publishable` check will fail until #289 lands, and this branch is rebased)

**Comment by mzdm** [comment]
> (The `publishable` check will fail until #289 lands, and this branch is rebased)

Alright, thank you for checking this and making the other changes!

I see `local_tests CHANNEL:stable` CI check is stuck and queued for +8hrs

**Comment by stuartmorgan** [comment]
289 has landed, so you can merge in master and the test will pass.

**Comment by ditman** [comment]
Merged and published https://pub.dev/packages/pointer_interceptor/versions/0.9.0

It's still being listed as "Preview", I guess because of the version of the SDK we used being a "-0". We might have to revisit this to move 0.9.0 to stable.

**Comment by stuartmorgan** [comment]
Preview is different than Pre-release; it just means that Pub has noticed that it depends on a version of the Dart SDK that's not yet released to stable (in this case, 2.12). It should automatically change to being listed as stable once that happens (per the very end of [this recent post](https://medium.com/dartlang/preparing-the-dart-and-flutter-ecosystem-for-null-safety-e550ce72c010)).

**Comment by ditman** [comment]
Ooooh, cool, I missed that memo! I thought the publishing warning of "you're publishing a not-preview version that depends on a preview version of the SDK" had toggled something when I published. Thanks for the link!

**Pull request merged**
